sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -564,12 +564,12 @@ public static <T> T getResultWithTimeout(Future<T> future, Duration timeout)
         throws InterruptedException, ExecutionException, TimeoutException {
         Objects.requireNonNull(future, "'future' cannot be null.");
 
-        if (!hasTimeout(timeout)) {
+        if (timeout == null) {
             return future.get();
         }
 
         try {
-            return future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
+            return ImplUtils.getResultWithTimeout(future, timeout.toMillis());
         } catch (TimeoutException e) {
             future.cancel(true);
```

Comment:
Now that the ImplUtils handles the timeout, wouldn't this cancel the future twice and catch the TimeoutException twice?

Reply:
Good catch, will fix that